### ISSUE TITLE
problem of " Error detected in AddmmBackward" in training

### ISSUE THREAD

**Issue by aiaini66** [Juyong/CaricatureFace#36]
When training to 500 epochs, the following error is reported because the vertices used to calculate the keypoint loss use the target:

![Screenshot from 2022-02-17 10-25-18](https://user-images.githubusercontent.com/27721212/154393232-410187a7-01d5-4053-8494-4fe19e2c640b.png)

What is the reason for this and how should I deal it?

**Comment by aiaini66** [comment]
@Juyong  I‘m looking forward to your reply!

**Comment by aiaini66** [comment]
@RainbowRui  I‘m looking forward to your reply!

**Comment by RainbowRui** [comment]
Do you use our recommended version of PyTorch?

**Comment by aiaini66** [comment]
@RainbowRui  i used pytorch1.4, but cuda is 11.2, i guess  it is a version problem of cuda？

**Comment by aiaini66** [comment]
@RainbowRui @Juyong  I want to render the vertex values predicted by the network to the 2d plane, but I found that the magnitude of the trans matrix from model_1 is very different from the magnitude of the vertices obtained from model_2. How should the vertex values from model_2 be processed to match the trans matrix? The order of magnitude is consistent, or how do you normalize the vertex values of each cartoon graph? 
![Screenshot from 2022-02-21 14-36-17](https://user-images.githubusercontent.com/27721212/154901806-bd3d8fcc-35ea-4118-9be4-0d7361fdde26.png)
hope you reply. thanks!!!

**Comment by RainbowRui** [comment]
We use weak perspective projection in our method, and you can refer to Formula 9 in our paper.

**Comment by aiaini66** [comment]
Thank you very much !!!
